feat(add/edit employee): Add/edit employee page with image upload.

ashwin8448 opened the pull request:
## Description
This pull request introduces an employee add/edit page with functionalities to enhance the overall user experience. The key features include:

1. **Multi-page form with progress bar:**
   
   - Implemented a multi-page form with a progress bar.
   - A review page so that details can be reviewed before submission.

2. **Image upload:**
   
   - Employee image can be added/modified.
   
3. **Validation**

   - All form fields are validated.
   - User can move from one-section to another section only if all form fields are valid.

4. **Redux Integration:**

   - Configured the Redux store to manage the global state.
   - Implemented core functions for fetching data via Axios using Redux.
   - Set up actions and reducers to manage data flow and state for a more robust architecture.

ashwin8448 committed:
[edit]: View employee page

Commit:
[edit]: Styling view employee page

Commit:
[add]: Added old form structure

ashwin8448 committed:
Merge branch 'components' of github.com:ashwin8448/HRM_Redux into viewEmployee

ashwin8448 committed:
[edit]: Added active chip and image in employee listing page.

ashwin8448 committed:
Merge branch 'viewEmployee' of github.com:ashwin8448/HRM_Redux into formPage

ashwin8448 committed:
[edit]: Form page

ashwin8448 committed:
[edit]: Added previous form in form page

ashwin8448 committed:
[edit]: Minor changes in form page

Commit:
[Add]: Added progress bar in form

ashwin8448 committed:
[edit]: Form page

ashwin8448 committed:
[edit]: Removed comments

Commit:
edit:[form pag]

Commit:
[edit]: Photo upload

ashwin8448 committed:
[edit]: Form page

Commit:
[add]: form config

Commit:
[edit]: form validation

ashwin8448 committed:
[Edit]: Photo upload.

ashwin8448 committed:
[Edit: User image upload]

Commit:
[edit]:photo upload

ashwin8448 committed:
[edit]: Interface, form page and form submission

Commit:
[edit]: Added new interfaces to manage data

Commit:
[edit]:Removed unused components

Commit:
[fix]: Skillchip bug

Commit:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into formPage

Commit:
[edit]: API Interceptor

Commit:
[edit]: View Employee page

Commit:
[edit]: Deleted unused filses

Commit:
[delete]: Unused files

ashwin8448 committed:
Update deploy.yml

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+  const [activeSection, setActiveSection] = useState(1);
+  const [employeeData, setEmployeeData] = useState<IAppEmployee>();
+  //TODO: try to use dispatch
+  const dispatch = useDispatch();
```
remove unused dispatch

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+
+  useEffect(() => {
+    if (employeeData)
+      for (let ObjKey in employeeData) {
```
use specific variable instead of objkey

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+    }
+  });
+  if (isLoading) return <Loader className="center-screen" />;
+  const formConfig = [
```
move this to a configuration file

seraGeorge commented on src/components/Details/Details.tsx:
```diff
@@ -0,0 +1,32 @@
+    <DetailsWrapper $newline={newline} $skill={typeof content != "string"}>
+      <div className="description">
+        <span className="material-symbols-outlined ">{icon}</span>
+        {title && <span className="title">{title}</span>}
```
if no icon then skip the margin left

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+                  </EmployeeViewWrapper>
+                </Fieldset>
+                <ButtonGrpWrapper>
+                  <Button icon="" onClick={() => setActiveSection(1)}>
```
use edit button icon instead of this edit button on the top right of the page.. near the review

seraGeorge commented on src/pages/EmployeeView/EmployeeView.tsx:
```diff
@@ -0,0 +1,115 @@
+          />
+          <DetailsSection
+            title="Work Experience"
+            content={getWorkExp(employee.dateOfJoining!)}
```
if work experience < 1 month, set text as less than 1 month

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+            )}
+            {activeSection < 4 && (
+              <ButtonGrpWrapper>
+                {/* TODO: disabled button */}
```
remove todo

seraGeorge commented on src/pages/EmployeeUpdate/Form.tsx:
```diff
@@ -0,0 +1,394 @@
+    try {
+      if (urlType === "add-employee") {
+        const newEmployee = await convertFormDataToIPostEmployees(formValues);
+        await postData(apiURL.employee, newEmployee);
```
error adding employee

ashwin8448 committed:
[edit]: Changes made as per review

seramg committed:
Merge branch 'develop' of github.com:ashwin8448/HRM_Redux into formPage

seramg committed:
[add]: changes to redux

seraGeorge committed:
[add]: color change for disabled btn
[add]: key for employee name list item
[add]: status range to accomodate 201 status
[add]: skillsdata,deptdata,rolesdata for actions
[add]: setloading type action
[remove]: store.dispatch and used dispatch
[remove]: aliasing for action.payload
[add]: action type in reducers
[add]: useAppSelector instead of useSelector
[remove]: unnecessary props
[add]:link to the edit employee
[add]: navigation after adding employee
[add]: const instead of let in for loop
[remove]: border in view employee
[remove]: unnecessary div in view employee layout
